File: test/fixtures/sample-sdk-elixir/lib/workos/organizations/organization.ex
defmodule WorkOS.Organizations.Organization do
  @moduledoc """
  Struct representing an Organization.
  """

  @type t :: %__MODULE__{
    id: String.t(),
    name: String.t(),
    status: String.t(),
    allow_profiles_outside_organization: boolean(),
    domains: [String.t()] | nil,
    created_at: String.t(),
    updated_at: String.t()
  }

  defstruct [:id, :name, :status, :allow_profiles_outside_organization, :domains, :created_at, :updated_at]
end

defmodule WorkOS.Organizations.Status do
  @moduledoc """
  Enum for organization status.
  """

  @type t :: :active | :inactive

  def active, do: "active"
  def inactive, do: "inactive"

  def values, do: ["active", "inactive"]
end

defmodule WorkOS.Organizations.Order do
  @moduledoc """
  Enum for ordering.
  """

  @type t :: :asc | :desc

  def asc, do: "asc"
  def desc, do: "desc"

  def values, do: ["asc", "desc"]
end

defmodule WorkOS.Organizations.ListOrganizationsResponse do
  @moduledoc """
  Struct representing a list organizations response.
  """

  @type t :: %__MODULE__{
    data: [WorkOS.Organizations.Organization.t()],
    list_metadata: WorkOS.Common.ListMetadata.t()
  }

  defstruct [:data, :list_metadata]
end

defmodule WorkOS.Organizations.CreateOrganizationOptions do
  @moduledoc """
  Options for creating an organization.
  """

  @type t :: %__MODULE__{
    name: String.t(),
    domains: [String.t()] | nil
  }

  defstruct [:name, :domains]
end

File: test/fixtures/sample-sdk-dotnet/src/WorkOS/Organizations/Organization.cs
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace WorkOS.Organizations
{
    [JsonConverter(typeof(StringEnumConverter))]
    public enum OrganizationStatus
    {
        [EnumMember(Value = "active")]
        Active,

        [EnumMember(Value = "inactive")]
        Inactive,
    }

    [JsonConverter(typeof(StringEnumConverter))]
    public enum Order
    {
        [EnumMember(Value = "asc")]
        Asc,

        [EnumMember(Value = "desc")]
        Desc,
    }

    public class Organization
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("status")]
        public OrganizationStatus Status { get; set; }

        [JsonProperty("allow_profiles_outside_organization")]
        public bool AllowProfilesOutsideOrganization { get; set; }

        [JsonProperty("domains")]
        public List<string> Domains { get; set; }

        [JsonProperty("created_at")]
        public string CreatedAt { get; set; }

        [JsonProperty("updated_at")]
        public string UpdatedAt { get; set; }
    }

    public class GetOrganizationOptions
    {
        [JsonProperty("organization")]
        public string Organization { get; set; }
    }

    public class ListOrganizationsOptions
    {
        [JsonProperty("limit")]
        public int? Limit { get; set; }

        [JsonProperty("order")]
        public Order? OrderBy { get; set; }

        [JsonProperty("after")]
        public string After { get; set; }
    }

    public class ListOrganizationsResponse
    {
        [JsonProperty("data")]
        public List<Organization> Data { get; set; }

        [JsonProperty("list_metadata")]
        public Common.ListMetadata ListMetadata { get; set; }
    }

    public class CreateOrganizationOptions
    {
        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("domains")]
        public List<string> Domains { get; set; }
    }
}
